pkg/validation/internal/multiarch.go

```diff
@@ -0,0 +1,513 @@
+// to inform where their container tool that should be used to inspect the image
+const ContainerToolsKey = "container-tools"
+
+// operatorFrameworkArchLabel defines the label used to the store the supported Arch on CSV
+const operatorFrameworkArchLabel = "operatorframework.io/arch."
+
+// operatorFrameworkSoLabel defines the label used to the store the supported So on CSV
+const operatorFrameworkSoLabel = "operatorframework.io/so."
+
+// MultiArchValidator store the data to perform the tests
+type MultiArchValidator struct {
+	// InfraArchLabels store the arch labels (i.e amd64, ppc64le) from
+	// operatorframework.io/arch.<GOARCH>: supported
+	InfraArchLabels []string
+	// InfraSOLabel store the OS labels from
+	// operatorframework.io/so.<GOARCH>: supported
+	InfraSOLabels []string
+	// Images stores the images defined in the bundle with the platform.arch supported
+	Images map[string][]Platform
+	// AllArchFound contains a map of the arch types found
+	AllArchFound map[string]string
+	// AllOsFound contains a map of the so(s) found
+	AllOsFound map[string]string
+	// Store the bundle load
+	Bundle *manifests.Bundle
+	// ContainerTool defines the container tool which will be used to inspect the images
+	ContainerTool string
+	// Warns stores the errors faced by the validator to return the warnings
+	Warns []error
+}
+
+// ManifestInspect store the data obtained by running container-tool manifest inspect <IMAGE>
+type ManifestInspect struct {
+	ManifestData []ManifestData `json:"manifests"`
+}
+
+// ManifestData store the platforms
+type ManifestData struct {
+	Platform Platform `json:"platform"`
+}
+
+// Platform store the Architecture and OS supported by the image
+type Platform struct {
+	Architecture string `json:"architecture"`
+	OS           string `json:"os"`
+}
+
+func multipleArchitecturesValidate(objs ...interface{}) (results []errors.ManifestResult) {
+	// Obtain the k8s version if informed via the objects an optional
+	var containerTool = ""
+	for _, obj := range objs {
+		switch obj.(type) {
+		case map[string]string:
+			containerTool = obj.(map[string]string)[ContainerToolsKey]
+			if len(containerTool) > 0 {
+				break
+			}
+		}
+	}
+
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateMultiArchWith(v, containerTool))
+		}
+	}
+	return results
+}
+
+func validateMultiArchWith(bundle *manifests.Bundle, containerTool string) errors.ManifestResult {
+	result := errors.ManifestResult{}
+	if bundle == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle is nil", nil))
+		return result
+	}
+
+	result.Name = bundle.Name
+
+	if bundle.CSV == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle csv is nil", bundle.Name))
+		return result
+	}
+
+	containerTool, err := validateContainerTool(containerTool)
+	if err != nil {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+		return result
+	}
```

Comment:
I think we should move this block up to `multipleArchitecturesValidate`. This function does not actually use `containerTool`, it just accepts it as a parameter and passes it into the validator. In my view, the function that retrieves the `containerTool` value should be the one to validate it and return an error.

Reply:
The idea is valid for all input first. We could have other keys. 
So,  would not make sense to move forward without first checking if the input is correct. 
See that if we do what you suggest we will start to do the checks before ensuring that it is actually possible with the data provided.

Reply:
So, I tried to clarify in the code and I am closing this one as solved let me know if you think that we should re-open.